### ISSUE TITLE
When upgrading scalafmt, run scalafmt on the codebase

### ISSUE THREAD

**Issue by Daenyth** [scala-steward-org/scala-steward#1056]
I think it makes sense that when scalafmt gets upgraded, steward should run it on the codebase after update. I'd also suggest a configuration toggle that repositories can add to their steward config in git that will disable that behavior, for people who prefer formatter updates to be executed as-needed instead of all files at once

**Comment by sloshy** [comment]
I would love to see this feature. I'm not entirely sure about it being the default, though, for two reasons. One being that scalafmt changes mean merge conflicts with anything and everything that is currently in the development pipeline, two being, a change to default behavior for everybody signed up on the project at this point that goes beyond heuristics seems like it could be potentially off-putting to some. Definitely a good thing to think about though.

**Comment by manuelcueto** [comment]
Considering that people who use scalafmt usually run it on compile or before creating a PR, if you were in a development branch and rebase against a master with scalafmt updated you'd probably run it anyway. don't think it would be extremely disruptive.

**Comment by sloshy** [comment]
@manuelcueto IME format-on-compile is the source of a lot of merge conflicts and doesn't scale well with projects that have lots of active contributors. It is also no longer recommended by scalafmt's own documentation: https://scalameta.org/scalafmt/docs/installation.html#format-on-compile

The recommendation is to use format-on-save and get eventually consistent formatting throughout your project. Not only does that mean less merge conflicts but it also means that your editor does not get confused when the in-memory layout of changes you have pending is no longer compatible with what's stored on the disk. Happened to me all the time in IntelliJ before I switched over to format-on-save (and now I use Metals which does this by default). It also means there are less weird whitespace changes to mentally filter through, making scalafmt less of a hassle to review with.

I think that the option for it to be part of a scala-steward request is nice to have, though I would not personally use it at my job. It would be quite nice for a low-traffic project.

**Comment by robstoll** [comment]
@fthomas I can provide a PR for this behaviour as we use it at @tegonal. However, we have not implemented a config for this, scalafmt runs always in case of a scalafmt update. Let me know if you would like to introduce a new config flag for this

**Comment by fthomas** [comment]
@robstoll I'd prefer a local flag in `.scala-steward.conf` which is disabled by default over a global configuration.

**Comment by robstoll** [comment]
@fthomas alright, shall I make the scalafmt commands configurable as well? We are running `sbt scalafmtAll scalafmtSbt`

**Comment by renatocaval** [comment]
If we add this, it's important to keep two commits. One with the scalafmt update and on with the code formatting. 

The reason being that many projects also have maintenance branches and we need to be able to only cherry-pick the scalafmt update without taking all the formatted files. This to avoid merge conflicts.

Then, on the maintenance branch, one will need to run the formatting by hand. 
Usually, maintenance branches have low activity and we don't want to keep updating dependencies, but in the case of formatting, we do need to keep them in sync so bug fix backports don't generate conflict because of formatting changes.

**Comment by kiranbayram** [comment]
Hello @robstoll , are you still working on this ?

**Comment by robstoll** [comment]
No, as I did not get feedback on #1903 on which this one would have been based on. However we are using a working implementation (which is not based on master though).

**Comment by kiranbayram** [comment]
Thanks for the info @robstoll 👍

**Comment by kluen** [comment]
I was wondering if running scalafmt only after scalafmt updates is really the best option.
In general, we want to format whenever formatting would be needed after any update.
There are at least two more cases I can think of:
- when Scalafix rewrites code as part of an update
- if you use alignment, the dependencies in `build.sbt` might need formatting after any update

If such a flag is implemented, it would be great being able to configure it to always run scalafmt.
